### ISSUE TITLE
Change to access protocol for firmware versions V3316-2008 and higher

### ISSUE THREAD

**Issue by MorganAskins** [sergey-inform/SIS3316#1]
As outlined in the updated Ethernet manual for the 3316, the access protocol now provides space for a 1 byte word to reference the packet identifier. This byte goes between the Acknowledge/Request byte and the address word when sending/receiving a message. Ideally the software should do the following.
- Check for firmware version before sending a message
- For newer firmware insert the packet identifier bit
- Keep track of packet identifiers, increment them, and use them to catch/avoid packet loss

Any thoughts? Inserting a packet identifier into the byte string is simple and allows the software to work with the newer firmware; however, I don't use the packet identifier for anything at the moment. Have you noticed any problems with a mismatch in call/response without this "feature" that struck has added?

**Comment by sergey-inform** [comment]
Sounds great! 
If only they add a checksum for the data, maw-data length field and a couple of extra features to make the access protocol more robust...

Actually we are still using the previous firmware version in our setup (at least until the summer), and a half-duplex 500Mbps mode (without chaining the requests) is satisfactory for us. I had to add a TCP-like collision avoidance algorithm to handle packet losses and prevent network congestion.

Since we can not find another SIS3316 unit for less than €15 555 in Russia, and we need at least two units for a new setup, not sure I'll check out the new features soon.

**Comment by dougUCN** [comment]
As of 4edafd2aa967a7f97ffdd0d11f08523becb66843 VME V3316-2008 and higher is now supported.

The packet identifier in the byte string is more or less useless. According to the ethernet manual packets are only dropped if you're on Windows. They also implemented a packet re-listen command but it is only useful for catching _single_ packet call/response. This is useless for the FIFO read protocols (where most problems are likely to occur) when multiple packets are being sent over 

Sergey's TCP collision avoidance is a godsend. Most dropped packet issues from the sis3316 seem to be from network congestion, and his algorithm catches most problems. I've found that if you configure your MTU to enable jumbo frame settings, and enable the jumbo frame bits on the sis3316, then the increased read speed allows stability for a 100kHz input signal without much issue.

Anyway, in summary: the packet identifier is more or less useless, VME V3316-2008 and higher is now supported, and this issue can be closed.

**Comment by sergey-inform** [comment]
Ok.
_My apologies for the slow reply._
